p2p-sync/src/main/java/org/smartregister/p2p/view/QRCodeScannerView.java

```diff
@@ -0,0 +1,204 @@
+ */
+
+public class QRCodeScannerView extends LinearLayout implements Detector.Processor<Barcode>, LifecycleObserver {
+
+    private CameraSource cameraSource;
+    private CameraSourcePreview cameraSourcePreview;
+
+    private ArrayList<OnQRRecognisedListener> onQRRecognisedListeners = new ArrayList<>();
+
+    public QRCodeScannerView(Context context) {
+        super(context);
+        init();
+    }
+
+    public QRCodeScannerView(Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    public QRCodeScannerView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init();
+    }
+
+    private void init() {
+        inflate(getContext(), R.layout.view_barcode_scanner, this);
+
+        cameraSourcePreview = findViewById(R.id.csp_barcodeScannerView_preview);
+        createCameraSource();
+
+        // Register to the lifecycle
+        if (getContext() instanceof LifecycleOwner) {
+            LifecycleOwner lifecycleOwner = (LifecycleOwner) getContext();
+            lifecycleOwner.getLifecycle().addObserver(this);
+        }
+    }
+
+    /**
+     * Creates and starts the camera.  Note that this uses a higher resolution in comparison
+     * to other detection examples to enable the barcode detector to detect small barcodes
+     * at long distances.
+     * <p>
+     * Suppressing InlinedApi since there is a check that the minimum version is met before using
+     * the constant.
+     */
+    @SuppressLint("InlinedApi")
+    private void createCameraSource() {
+        Context context = getContext().getApplicationContext();
+        BarcodeDetector barcodeDetector = new BarcodeDetector.Builder(context)
+                .setBarcodeFormats(Barcode.QR_CODE)
+                .build();
+        barcodeDetector.setProcessor(this);
+
+        if (!barcodeDetector.isOperational()) {
+            Timber.w("Detector dependencies are not yet available.");
+            IntentFilter lowStorageFilter = new IntentFilter(Intent.ACTION_DEVICE_STORAGE_LOW);
+            boolean hasLowStorage = getContext().registerReceiver(null, lowStorageFilter) != null;
+
+            if (hasLowStorage) {
+                Toast.makeText(getContext(), "Face detector dependencies cannot be downloaded due to low device storage", Toast.LENGTH_LONG).show();
+                Timber.e("Face detector dependencies cannot be downloaded due to low device storage");
+            }
+        }
+
+        CameraSource.Builder builder = new CameraSource.Builder(getContext().getApplicationContext(), barcodeDetector)
+                .setFacing(CameraSource.CAMERA_FACING_BACK)
+                .setRequestedPreviewSize(1600, 1024)
+                .setAutoFocusEnabled(true)
+                .setRequestedFps(45.0f);
+
+        cameraSource = builder.build();
+    }
+
+    @Override
+    public void release() {
+        //Todo
```

Comment:
Implement this

Reply:
It is supposed to release any resources being used by the processor according to https://developers.google.com/android/reference/com/google/android/gms/vision/Detector.Processor.html#release()

These are resources being used in `receiveDetections`. The `cameraSource` to which the detector is passed calls the `release` method